topdown/cache/cache_test.go

```diff
@@ -165,6 +177,88 @@ func TestInsert(t *testing.T) {
 	}
 }
 
+func TestInterQueryValueCache(t *testing.T) {
+
+	in := `{"inter_query_builtin_value_cache": {"max_num_entries": 4},}`
+
+	config, err := ParseCachingConfig([]byte(in))
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	cache := NewInterQueryValueCache(context.Background(), config)
+
+	cache.Insert("foo", "bar")
+	cache.Insert("foo2", "bar2")
+	cache.Insert("hello", "world")
+	dropped := cache.Insert("hello2", "world2")
+
+	if dropped != 0 {
+		t.Fatal("Expected dropped to be zero")
+	}
+
+	_, found := cache.Get("foo")
+	if !found {
+		t.Fatal("Expected key \"foo\" in cache")
+	}
+
+	dropped = cache.Insert("foo3", "bar3")
+	if dropped != 1 {
+		t.Fatal("Expected dropped to be one")
+	}
+
+	_, found = cache.Get("foo3")
+	if !found {
+		t.Fatal("Expected key \"foo3\" in cache")
+	}
+
+	// update the cache config
+	in = `{"inter_query_builtin_value_cache": {"max_num_entries": 0},}` // unlimited
+	config, err = ParseCachingConfig([]byte(in))
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	cache.UpdateConfig(config)
+
+	cache.Insert("a", "b")
+	cache.Insert("c", "d")
+	cache.Insert("e", "f")
+	dropped = cache.Insert("g", "h")
+
+	if dropped != 0 {
+		t.Fatal("Expected dropped to be zero")
+	}
+
+	// at this point the cache should have 8 entries
+	// update the cache size and verify multiple items dropped
+	in = `{"inter_query_builtin_value_cache": {"max_num_entries": 6},}`
+	config, err = ParseCachingConfig([]byte(in))
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	cache.UpdateConfig(config)
+
+	dropped = cache.Insert("i", "j")
+
+	if dropped != 3 {
+		t.Fatal("Expected dropped to be three")
+	}
+
+	_, found = cache.Get("i")
+	if !found {
+		t.Fatal("Expected key \"i\" in cache")
+	}
+
+	cache.Delete("i")
+
+	_, found = cache.Get("i")
```

Comment:
Looks like no part of this test actually asserts that we can retrieve the expected value too.